server/src/main/java/org/apache/calcite/avatica/server/AvaticaJsonHandler.java

```diff
@@ -106,14 +106,18 @@ public void handle(String target, Request baseRequest,
           // Avoid a new buffer creation for every HTTP request
           final UnsynchronizedBuffer buffer = threadLocalBuffer.get();
           try (ServletInputStream inputStream = request.getInputStream()) {
-            rawRequest = AvaticaUtils.readFully(inputStream, buffer);
+            byte[] bytes = AvaticaUtils.readFullyToBytes(inputStream, buffer);
+            String encoding = request.getCharacterEncoding();
+            if (encoding == null) {
+              encoding = "UTF-8";
```

Comment:
Instead of hardcoding "UTF-8" here wouldn't be better to obtain this information in more generic/configurable way. Maybe something along the lines of `Charset.defaultCharset();` or `System.getProperty("file.encoding")`?

Reply:
`StandardCharsets` would be a preferred way of getting a Charset instance.

Reply:
+1 for StandardCharsets.UTF_8, i think the default UTF8 is okey, cause UTF8 is the most popular encoding for Internet.

Reply:
@zabetak , @danny0405 , `request.getCharacterEncoding();` returns `String`, and I used `"UTF-8"` here just to simplify the code and have a single `new String(bytes, encoding)` call for both cases (encoding is set, and encoding is not set)

Reply:
If we always want to use `UTF-8` then we can even leave the code as is. I mentioned `defaultCharser` in the case that we want to allow other encodings that depend on the VM and the underlying OS.

Reply:
> If we always want to use UTF-8 then we can even leave the code as is

Yeah, we don't need to let people provide something else. UTF-8 will be sufficient to encode everything (afaik). Using some other encoding isn't worth the hassle of us altering the protocol to allow users to tell us that they want to use some other encoding :)